### ISSUE TITLE
VPN recognition seems broken/flappin

### ISSUE THREAD

**Issue by freifunk-en** [ffnord/mesh-announce#25]
Last month we did update all our communitys to use the lastest mesh-announce,
before we used alfred-announce from somewhere in 2015.

With the old alfred-announce we never had problems with announce/meshviewer to recognize vpn links.

Since we did install the new version with respondd it started flappin or not even show vpn links

![image](https://user-images.githubusercontent.com/16792249/32991827-022d4a00-cd43-11e7-9603-0d24dccdf5ef.png)

We´re using:
Batman-adv 2013.4 (from git)
Tunneldigger
Gluon 2016.x / 2017.x
mesh-announce
ffrgb-meshviewer
yanic

batman interface > br0
client tunnel > bat0
site2site tunnel > bat0

I´ve noticed that the entire structure inside nodeinfo.d has changed, but what could cause this problem?

Regards
Matthias

**Comment by jplitza** [comment]
Interesting. We have an identical setup except we're using fastd instead of Tunneldigger and never experienced these problems.

Whether links are seen as VPN links AFAIK entirely depends on whether one of the end nodes has the nodeinfo.vpn attribute set. So when the data from that one VPN gateway cannot be received, all its links are recognized as regular links, but then again, it shouldn't have a name either.

So basically, I suspect this is caused by the technology switch "Alfred → respondd" and not by mesh-announce and could only be mitigated by yanic. But maybe @genofire can say more about the problem.

**Comment by genofire** [comment]
@ yanic did not use the `nodeinfo.vpn` attribute anymore for vpn detection, because a other community has an gateway with a real interface to there mesh-network.

It use the mac address of `nodeinfo.mesh.{INTERFACE}.interfaces.tunnel` on both sides.
(Maybe we have an bug, if the both nodes has there mac on different types - `wifi` or `other`)

**Comment by jplitza** [comment]
@freifunk-en Is this still an issue with the fix in yanic?

**Comment by freifunk-en** [comment]
@jplitza Sorry for the late reply, i didnt update yanic yet but i will do today and report back

**Comment by AiyionPrime** [comment]
So, this issue basically is dead, as there are no users anymore, that can even confirm it?

**Comment by TobleMiner** [comment]
Does this issue still exist? There have been major changes in mesh-announce and thus additional confirmation on this issue is required. This issue will be closed in a month if there is no further activity.
